WolvenKit.RED4/Types/Primitives/Fundamentals/CInt16.cs

```diff
@@ -45,4 +45,22 @@ public override bool Equals(object? obj)
 
     public override string ToString() => _value.ToString();
     public string ToString(CultureInfo cultureInfo) => _value.ToString(cultureInfo);
+
+    #region IComparable, IComparable<CInt16>
+
+    public int CompareTo(object? value)
+    {
+        if (value == null)
+        {
+            return 1;
+        }
+        if (value is CInt16 i)
+        {
+            return CompareTo(i);
+        }
+        throw new ArgumentException("Value is not a CInt16", nameof(value));
+    }
+    public int CompareTo(CInt16 value) => _value - value._value;
```

Comment:
why does this work here and not for CInt32?

Reply:
Cannot use return (_value - value) as this causes a wrap around in cases where _value - value > MaxValue.

Reply:
right, btw would it make sense to just cast and deffer the implementation to the net types? or is there a problem with that? or is it not elegant?

Reply:
Yeah, could do that to. (Except for CBool)
Asked myself which would be better.
Just decided for reimplementing them to save one function call. No other reason.
Can change if you want.

Reply:
nope, was just curious